forge/src/main/java/net/blay09/mods/balm/forge/world/ForgeBalmWorldGen.java

```diff
@@ -76,6 +76,15 @@ public <T extends PlacementModifierType<?>> DeferredObject<T> registerPlacementM
         return deferredObject;
     }
 
+    @Override
+    public <T extends PoiType> DeferredObject<T> registerPoiType(ResourceLocation identifier, Supplier<T> supplier) {
+        return new DeferredObject<>(identifier, () -> {
+            T poiType = supplier.get();
+            Registry.register(BuiltInRegistries.POINT_OF_INTEREST_TYPE, identifier, poiType);
+            return poiType;
+        }).resolveImmediately();
```

Comment:
```suggestion
        DeferredRegister<PoiType> register = DeferredRegisters.get(ForgeRegistries.POI_TYPES, identifier.getNamespace());
        RegistryObject<T> registryObject = register.register(identifier.getPath(), supplier);
        return new DeferredObject<>(identifier,registryObject, registryObject::isPresent);
```
Shifting suggestion here for proper merging upon review